### DISCUSSION TITLE
Replace <div> with syntaxic HTML5 tags

### DISCUSSION THREAD

**Discussion by dascritch** [tabler/tabler#1383] [Ideas]
The interface is really cool, eye-candy, catchy, productive-oriented and you have set a very large lot of components. The rendered result is really good.
Except I find it too verbose : too many meaningless `<div>` and `<span>`, a lot of redundant class-names, and a lot of nested div into div into div.

I fear the actual verbosity will be too complex to maintain for my  team, and hard to maintain or leaky in a webapp.

So, here is my question : why not using syntaxic html, as `<header>`, `<aside>` tags and so on ? Why not trying to reduce the deep of html elements ? Bootstrap css used anonymous `<div>` because it was created before html5 was stabilized. And the main goal to HTML5 is to get again meaning tag names, identified sections, implicit accessibility, reduced nested code and easier maintenance. And  as we now have css reset values, css variables, css computations, `:is()`, and even [css layers](https://developer.mozilla.org/en-US/docs/Web/CSS/@layer) (I agree, the last one is brand new), we can be really more creative and with a simplier HTML and CSS syntax. If we accept to add some javascript, we can even use the Shadow-DOM feature from the WebComponent HTML5 standard to unclutter the needed tags to style elements. 

I'm sure you had excellent reasons to do those code-design choices for Tabler, and I probably miss more than one important point. 

But, if you accept, let's take as an example a drop-down menu. Here is your syntax : 

```html
<div class="dropdown-menu dropdown-menu-end dropdown-menu-arrow">
      <a href="#" class="dropdown-item">Status</a>
      <a href="#" class="dropdown-item">Profile</a>
      <a href="#" class="dropdown-item">Feedback</a>
      <div class="dropdown-divider"></div>
      <a href="settings" class="dropdown-item">Settings</a>
      <a href="singin" class="dropdown-item">Logout</a>
    </div>
```

Why not using the `<menu>` syntax ? Yes, we're adding a `<li>`, but the produced html source will be lighter and directly supported by a11y systems. And so we probably won't need to add a specific class to `<a>` children, even for `<hr>` elements. Or simplify CSS selectors via `menu.dropdown` and `a.dropdown`.
I know it means complexity in CSS, but we'll win a lighter and more comprehensive HTML.

```html
<menu class="dropdown dropdown-end dropdown-arrow">
      <li><a href="#" class="dropdown">Status</a></li>
      <li><a href="#" class="dropdown">Profile</a></li>
      <li><a href="#" class="dropdown">Feedback</a></li>
      <li><hr /></li>
      <li><a href="settings" class="dropdown">Settings</a></li>
      <li><a href="signin" class="dropdown">Logout</a></li>
</menu>
```

Sorry, that was my two cents.

**Comment by codecalm** [comment]
good idea! :)
